cluster/eks.go

```diff
@@ -274,65 +261,84 @@ func (c *EKSCluster) createAWSCredentialsFromSecret() (*credentials.Credentials,
+	c.modelCluster.EKS.CurrentWorkflowID = workflowID
+
+	err := c.modelCluster.Save()
+	if err != nil {
+		return errors.WrapIf(err, "failed to persist cluster to database")
+	}
+
+	return nil
+}
+
 // CreateCluster creates an EKS cluster with cloudformation templates.
 func (c *EKSCluster) CreateCluster() error {
 	c.log.Info("start creating EKS cluster")
 
-	awsCred, err := c.createAWSCredentialsFromSecret()
+	vpcTemplate, err := pkgEks.GetVPCTemplate()
 	if err != nil {
-		return errors.WrapIf(err, "failed to retrieve AWS credentials from secret")
+		return errors.WrapIfWithDetails(err, "failed to get CloudFormation template for VPC", "cluster", c.modelCluster.Name)
 	}
 
-	awsSession, err := session.NewSession(&aws.Config{
-		Region:      aws.String(c.modelCluster.Location),
-		Credentials: awsCred,
-	})
+	subnetTemplate, err := pkgEks.GetSubnetTemplate()
 	if err != nil {
-		return errors.WrapIf(err, "failed to create AWS awsSession")
+		return errors.WrapIfWithDetails(err, "failed to get CloudFormation template for Subnet", "cluster", c.modelCluster.Name)
 	}
 
-	// role that controls access to resources for creating an EKS cluster
-	eksStackName := c.generateStackNameForCluster()
-	iamStackName := c.generateStackNameForIAM()
-	sshKeyName := c.generateSSHKeyNameForCluster()
-
-	c.modelCluster.RbacEnabled = true
+	iamRolesTemplate, err := pkgEks.GetIAMTemplate()
+	if err != nil {
+		return errors.WrapIfWithDetails(err, "failed to get CloudFormation template for IAM roles", "cluster", c.modelCluster.Name)
+	}
 
-	c.log.Info("getting CloudFormation template for creating node pools for EKS cluster")
 	nodePoolTemplate, err := pkgEks.GetNodePoolTemplate()
 	if err != nil {
 		return errors.WrapIfWithDetails(err, "failed to get CloudFormation template for node pools", "cluster", c.modelCluster.Name)
 	}
 
-	c.log.Info("getting CloudFormation template for creating a eksSubnetModel")
-
-	subnetTemplate, err := pkgEks.GetSubnetTemplate()
-	if err != nil {
-		return errors.WrapIfWithDetails(err, "failed to get CloudFormation template for Subnet", "cluster", c.modelCluster.Name)
+	input := workflow.CreateClusterWorkflowInput{
+		CreateInfrastructureWorkflowInput: workflow.CreateInfrastructureWorkflowInput{
+			Region:                         c.modelCluster.Location,
+			OrganizationID:                 c.GetOrganizationId(),
+			SecretID:                       c.GetSecretId(),
+			SSHSecretID:                    c.GetSshSecretId(),
+			ClusterUID:                     c.GetUID(),
+			ClusterName:                    c.GetName(),
+			VpcID:                          aws.StringValue(c.modelCluster.EKS.VpcId),
+			RouteTableID:                   aws.StringValue(c.modelCluster.EKS.RouteTableId),
+			VpcCidr:                        aws.StringValue(c.modelCluster.EKS.VpcCidr),
+			ScaleEnabled:                   c.GetScaleOptions() != nil && c.GetScaleOptions().Enabled,
+			VpcCloudFormationTemplate:      vpcTemplate,
```

Comment:
Is this a workflow input? Or rather a constructor parameter for activities?  IMHO it should be the latter.

Reply:
If we pass the VPC CF template as a constructor parameter can be the activity retried with a modified CF template?

Reply:
Yes. Actually, your current implementation prevents retrying activities with modified CF templates as they will be part of the workflow history.